### ISSUE TITLE
no-useless-call: should it flag call & apply when used on arrow functions?

### ISSUE THREAD

**Issue by aubergine10** [eslint/eslint#6439]
Rule: http://eslint.org/docs/rules/no-useless-call

Should this rule also pick up on calling/applying arrow functions (as `call` and `apply` have no effect on them, their `this` is set in stone)...?

```javascript
this.msg = 'bar'
const foo = ()=> `foo${this.msg}`;
foo.call({msg:'quux'}); // 'foobar'
```

That sort of thing caught me out when first using arrow functions, because I didn't realise _'lexical `this`'_ meant _'immutable lexical `this`'_

**Comment by platinumazure** [comment]
Being able to know for sure that a variable points to any value of a particular type, let alone an arrow function, is pretty difficult to do reliably with static analysis.

I like the idea-- I just don't know how feasible it is. There's been some discussion on whether escope or similar tools could be augmented to support limited type detection, but that hasn't gone much of anywhere.

**Comment by alberto** [comment]
Closing because of the reasons @platinumazure exposed.
